### PR TITLE
KMP in JavaScript was added

### PR DESCRIPTION
**Pull request by bohemian98**
Fixes #2555 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added KMP algorithm written in JavaScript.

#### Languages Used: 
- JavaScript

#### Files Added:
-  KMP.js

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Knuth_Morris_Pratt_Algorithm/KMP.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            {i=i+1;}
+        }
+    }
+}
+var a="abcabaabcabac";
+var b="abaa";
+
+var index=KmpPatternSearch(b,a);
+console.log("Pattern observed at index:",index);
+//Output will be 3
```
Comment (MastersAbh) [inline Knuth_Morris_Pratt_Algorithm/KMP.js:75]: Give a blank line at the end of file

File: Knuth_Morris_Pratt_Algorithm/KMP.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+            if(j!=0)
+            {j=lps[j-1];}
+            else
+            {i=i+1;}
+        }
+    }
+}
+var a="abcabaabcabac";
+var b="abaa";
```
Comment (MastersAbh) [inline Knuth_Morris_Pratt_Algorithm/KMP.js:71]: Pls take dynamic inputs

File: Knuth_Morris_Pratt_Algorithm/KMP.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            //Pattern ans text matched
+            i=i+1;
+            j=j+1;
+        }
+        if(j===len_pattern)
+        {  
+            return i-j;
+            j=lps[j-1];
+        }
+        else if(j!=len_pattern && pattern.charAt(j)!=text.charAt(i))
```
Comment (MastersAbh) [inline Knuth_Morris_Pratt_Algorithm/KMP.js:60]: space after if throughout the file

File: Knuth_Morris_Pratt_Algorithm/KMP.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        {
+
+            if(j!=0)
+            {j=lps[j-1];}
```
Comment (MastersAbh) [inline Knuth_Morris_Pratt_Algorithm/KMP.js:64]: move the inner block to next line

File: Knuth_Morris_Pratt_Algorithm/KMP.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    len_text=text.length;
+    
+    //We find Longest Proper Prefix to decide which chater will be matched next
+    var lps=new Array(len_pattern);
+    
+    lpsValues(lps,pattern,len_pattern);
+
+    //iterator
+    var i=j=0;
+    while(i<len_text)
```
Comment (MastersAbh) [inline Knuth_Morris_Pratt_Algorithm/KMP.js:47]: space after while across the file

**Comment by bohemian98** [comment]
@MastersAbh  I have made the changes suggested by you. Please review my PR.

**Comment by MastersAbh** [comment]
Hi @bohemian98 , I see 12 files in your commit. please remove those files as they are not needed in this PR.

**Comment by bohemian98** [comment]
Hello @MastersAbh I have deleted the files which weren't required. Please review my PR.

**Comment by bohemian98** [comment]
> <@UU5DN4G1X> please review my pr #2801 for issue #2555. I have made changes suggested by you.
<sub>[View in Slack](https://gssoc20.slack.com/archives/CUEQKLA5U/p1588145224045300)</sub>

**Comment by bohemian98** [comment]
@MastersAbh I have squashed commits. Please merge my PR if everything looks fine.

**Pull request merged**